tools/coqdep/lib/dune

```diff
@@ -1,5 +1,16 @@
 (library
  (name coqdeplib)
+ (public_name coq-core.coqdeplib)
```

Comment:
because coqdeplib now depends on coq-core/META it needs to also be in coq-core, otherwise build -p coq-stdlib cannot build dune_rule_gen

Reply:
I don't know a way to say we depend on package coq-core's META regardless of whether it's installed or locally built (`%{lib:coq-core:META}` could be a sane syntax for it but currently `%{lib}` only works for libraries not packages AFAICT)

Reply:
Does hackery like `%{lib:coq-core:../META}` work? (I expect not)

Reply:
I don't remember all the details, but in the end I failed to construct something like that using the rule language.

It is easy to do in OCaml tho (so indeed, Dune for Coq does correctly depend on the METAs that are relevant under `_build/install`)

Reply:
cc #19899